webapp/src/types/store.d.ts

```diff
@@ -0,0 +1,25 @@
+import {GlobalState as ReduxGlobalState} from 'mattermost-redux/types/store';
+
+export type GlobalState = ReduxGlobalState & {
+    'plugins-github': {
+        connected: boolean,
+        enterpriseURL: string,
+        organization: string,
+        username: string,
+        userSettings: UserSettingsData,
```

Comment:
Can we have this be based on the reducers directly? What is your opinion on this?

Reply:
@mickmister Can you please elaborate on this? Not really able to understand what you mean here.

Reply:
@Kshitij-Katiyar Instead of having all of the fields listed out here, and potentially get out of sync with the reducer, we can do something like this:

```suggestion
import reducer from './reducer';

export type GlobalState = ReduxGlobalState & {
    'plugins-github': typeof reducer;
}
```